### PR TITLE
Do not append .git extension during normalization for Azure hosted git (issue #643)

### PR DESCRIPTION
**Pull request by jessesuen**

### CODE REVIEW

**Review by alexmt — APPROVED** [review]

File: util/git/git.go [hunk 1]
```diff
@@ -43,20 +43,24 @@ func IsTruncatedCommitSHA(sha string) bool {
...
-
-	// process
 	repoURL, err := url.Parse(repo)
 	if err != nil {
 		return ""
 	}
-
-	// postprocess
+	hostname := repoURL.Hostname()
+	if !strings.HasSuffix(hostname, "dev.azure.com") && !strings.HasSuffix(hostname, "visualstudio.com") {
```
Comment (merenbach) [inline util/git/git.go]: @jessesuen wondering if intent might be more clear factoring out into an `isAzureDomain()` function.

File: util/git/git.go [hunk 2]
```diff
@@ -43,20 +43,24 @@ func IsTruncatedCommitSHA(sha string) bool {
...
+		// NOTE: not all git services support `.git` appended to their URLs (e.g. azure) so this
+		// normalization is not quite safe.
+		repo = ensureSuffix(repo, ".git")
+		repoURL, err = url.Parse(repo)
```
Comment (merenbach) [inline util/git/git.go]: Why are we re-parsing after adding the `.git` suffix? I'm having some difficulty coming up with a URL path that becomes invalid as a URL once a suffix is added.

Comment (jessesuen) [reply util/git/git.go]: It's not about URL parsing/validation. We need to append the git suffix conditionally, only after we determine the hostname/domain is not azure. So the first parse was used to allow us to extract the domain. The second is done after we determine it's safe to append the suffix.

**Comment by merenbach** [comment]
Right!  With regard to a separate function, I’m suggesting not hardcoding
the URLs in this function, but ratehr in a primitive that can be tested and
maintained independently, that returns true for an Azure URL, false
otherwise. Then in the function here, there is just one call to that. Seems
to me like the sort of check that benefits from its own testing unit, so
that the function here doesn’t need to change in future if Azure updates or
changes its URLs.

**Comment by merenbach** [comment]
The function also serves the purpose of self-documenting code since it’s
clear to future maintainers that here we’re checking isAzureGitURL, which
explains its intent withot a comment in the invoking function.

**Comment by jessesuen** [comment]
@merenbach i've created a helper function for this. PTAL

**Comment by merenbach** [comment]
LGTM!

**Pull request merged**